examples/localstack/localstack.go

```diff
@@ -0,0 +1,190 @@
+package localstack
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/docker/go-connections/nat"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/internal/testcontainersdocker"
+	"github.com/testcontainers/testcontainers-go/wait"
+	"golang.org/x/mod/semver"
+)
+
+const defaultPort = 4566
+const defaultRegion = "us-east-1"
+const defaultVersion = "0.11.2"
+const hostnameExternalEnvVar = "HOSTNAME_EXTERNAL"
+
+const defaultAccessKeyID = "accesskey"
+const defaultSecretAccessKey = "secretkey"
+const defaultToken = "token"
+
+// LocalStackContainer represents the LocalStack container type used in the module
+type LocalStackContainer struct {
+	testcontainers.Container
+	Credentials     Credentials
+	Region          string
+	EnabledServices map[string]Service
+}
+
+type Credentials struct {
+	AccessKeyID     string
+	SecretAccessKey string
+	Token           string
+}
+
+// ServicePort returns the port of the given service
+func (l *LocalStackContainer) ServicePort(ctx context.Context, service EnabledService) (nat.Port, error) {
+	if _, ok := l.EnabledServices[service.Name()]; !ok {
+		return "", fmt.Errorf("service %s is not enabled", service.Name())
+	}
+
+	internalPort := l.EnabledServices[service.Name()].servicePort()
+
+	p, err := nat.NewPort("tcp", fmt.Sprintf("%d", internalPort))
+	if err != nil {
+		return "", err
+	}
+
+	return l.MappedPort(ctx, p)
+}
+
+func runInLegacyMode(version string) bool {
```

Comment:
As discussed online, we should remove the support for legacy mode: it's a new module and we would like to define the minimal version that is supported, so no legacy mode should be possible.

It will simplify the code significantly

Reply:
Another option would be to keep it, and create an issue for future deprecation of the legacy mode, and notify consumers with semver